segmenter: Fix tests.

j0sh opened the pull request:
This feels a bit like a band-aid  -- if it's failing before writing the headers, is it really disconnecting "mid-stream" ? --  but seems to make the tests pass for now.

Fixes https://github.com/livepeer/lpms/issues/137
Fixes https://github.com/livepeer/lpms/issues/136

darkdarkdragon commented on segmenter/video_segmenter_test.go:
```diff
@@ -431,8 +431,8 @@ func (s *ServerDisconnectStream) ReadRTMPFromStream(ctx context.Context, dst av.
 	header, err := file.Streams()
-	dst.WriteHeader(header)
 	dst.Close()
+	dst.WriteHeader(header)
```
With this change this function as good as this:
```golang
func (s *ServerDisconnectStream) ReadRTMPFromStream(ctx context.Context, dst av.MuxCloser) (chan struct{}, error) {
	dst.Close()
	return make(chan struct{}), nil
}
```

Still, this doesn't test what it implies - it just closes connection before ffmpeg has chance to start processing actual packets 

> Curious, is the disconnect clean TCP/RTMP-wise?

For RTMP, I didn't found any 'closing' sequence in protocol description, so there is no 'unclean' disconnect RTMP-wise.

>  Is there a way we can hard-kill the server and have the client infer the disconnect, rather than having the connection tear itself down? That would capture the intent of this test better.

I did some research but wasn't able to find way to force TCP reset on connection from golang code.

j0sh replied:
Good point, changed the function to just close immediately.

> Still, this doesn't test what it implies - it just closes connection before ffmpeg has chance to start processing actual packets

If we rename the test to remove the reference to "midstream" then that fits the fixed behavior of the test. tcpdump confirms the connection is established and we get the expected I/O error. How does that look?

j0sh commented:
Renamed the test itself to `TestServerDisconnect` so the scenario under test fits the description better.  It's not necessarily "mid-stream" but still tests a disconnect and results in an IO error as expected; the test itself doesn't change, just the name and how we're stubbing out the failures.

Rebased to master and addressed @darkdarkdragon comments. https://github.com/livepeer/lpms/commit/646b45763b0cabc77cdaa5d0aac33bc00467e817
